### PR TITLE
Jenkins workflow support

### PR DESCRIPTION
**Pull request by lptr**
Added support for Jenkins Workflow, according to this manual:

https://github.com/jenkinsci/workflow-plugin/blob/master/basic-steps/CORE-STEPS.md#adding-support-from-plugins

Not sure what to do with the matrix project dependency. How are dependencies on other plugins handled in Jenkins plugins in general? Is this okay?

### CODE REVIEW

File: src/main/java/hudson/plugins/descriptionsetter/DescriptionSetterPublisher.java [hunk 1]
```diff
@@ -62,16 +64,25 @@ public BuildStepMonitor getRequiredMonitorService() {
...
-				&& build.getResult().isWorseThan(Result.UNSTABLE);
-		return DescriptionSetterHelper.setDescription(build, listener,
+				&& run.getResult().isWorseThan(Result.UNSTABLE);
+		DescriptionSetterHelper.setDescription(run, listener,
 				useUnstable ? regexpForFailed : regexp,
 				useUnstable ? descriptionForFailed : description);
 	}
 
+	@Override
+	public boolean perform(AbstractBuild<?, ?> build, Launcher launcher,
```
Comment (jglick) [inline src/main/java/hudson/plugins/descriptionsetter/DescriptionSetterPublisher.java]: This overload can simply be deleted.

File: pom.xml [hunk 2]
```diff
@@ -26,6 +26,14 @@
     <tag>HEAD</tag>
   </scm>
 
+  <dependencies>
+      <dependency>
+          <groupId>org.jenkins-ci.plugins</groupId>
+          <artifactId>matrix-project</artifactId>
+          <version>1.4.1</version>
```
Comment (jglick) [inline pom.xml]: Use 1.4 unless you are depending on a core version with the SECURITY-125 fix (so 1.596.1+).

File: src/main/java/hudson/plugins/descriptionsetter/DescriptionSetterBuilder.java [hunk 3]
```diff
@@ -34,8 +34,9 @@ public DescriptionSetterBuilder(String regexp, String description) {
 	public boolean perform(AbstractBuild<?, ?> build, Launcher launcher,
 			BuildListener listener) throws InterruptedException {
```
Comment (jglick) [inline src/main/java/hudson/plugins/descriptionsetter/DescriptionSetterBuilder.java]: This could probably be made a `SimpleBuildStep` too.

**Comment by jglick** [comment]
Note that as of [1.4](https://github.com/jenkinsci/workflow-plugin/blob/master/CHANGES.md#14-mar-16-2015) you can just write

```groovy
currentBuild.description = 'whatever'
```

There may still be value in integrating this plugin if it does some special computation of the description that would be awkward to replicate in your script.

**Comment by lptr** [comment]
Oh, I didn't know about `currentBuild`, thanks! That solves all my problems. Do you think it's still worth fixing this plugin?

**Comment by jglick** [comment]
> Do you think it's still worth fixing this plugin?

Not sure. I only know from glancing at its source code what all it does. Seems it has some facility to set a description according to parsing of the log text. Potentially useful, though I think this might be better done by giving you access to the log in `currentBuild` (ideally more efficiently than loading a multimeg `String` into memory!), or maybe a custom Workflow step to parse the current build’s log for a supplied regexp, etc. Then you could base decisions on that information using Groovy code. Depends on what people’s use cases are here.

**Comment by lptr** [comment]
For my use-case `currentBuild.description` is already enough functionality. I think exposing the log would be great, too. The only reason I started tempering with this one is that I haven't found documentation on `currentBuild`. Now searching for it I found it being mentioned in CORE-STEPS.md, but maybe some more explicit mention would be nice, like putting listing the description setter plugin in COMPATIBILITY.md with a note saying that you can use `currentBuild.description` instead.

**Comment by jglick** [comment]
IIRC the documentation for `currentBuild` is currently in the inline help for the script textarea.

**Comment by jglick** [comment]
But yeah, other mentions would be appropriate.

**Pull request closed without merging**